src/xenia/debug/debugger.cc

```diff
@@ -134,67 +135,70 @@ void SendResponse(SOCKET client_socket, flatbuffers::FlatBufferBuilder& fbb,
 }
 
 void Debugger::PreLaunch() {
-  accept_thread_ = std::thread([this]() {
-    xe::threading::set_name("Debugger Server");
-
-    while (listen_socket_ != INVALID_SOCKET) {
-      sockaddr_in6 client_addr;
-      int client_count = sizeof(client_addr);
-      SOCKET client_socket_id =
-          accept(listen_socket_, reinterpret_cast<sockaddr*>(&client_addr),
-                 &client_count);
-      if (client_socket_id == INVALID_SOCKET) {
-        XELOGE("Failed to accept socket");
-        continue;
-      }
+  if (!accept_thread_running_) {
```

Comment:
If you do `if (accept_thread_running_) { return ; }` here then we can avoid indenting the whole function one level :)